android/BOINC/app/src/main/java/edu/berkeley/boinc/client/Monitor.kt

```diff
@@ -196,6 +216,40 @@ class Monitor : LifecycleService() {
         val offFilter = IntentFilter(Intent.ACTION_SCREEN_OFF)
         registerReceiver(screenOnOffReceiver, onFilter)
         registerReceiver(screenOnOffReceiver, offFilter)
+
+        // rpcExtern register the receiver
+        val intentFilter = IntentFilter()
+        intentFilter.addAction(NETWORK_STATE_CHANGED_ACTION)
+        intentFilter.addAction("RPC_EXTERN")
+        registerReceiver(mRpcExternBroadCastReceiver, intentFilter)
+//      LocalBroadcastManager.getInstance(this).registerReceiver(mRpcExternBroadCastReceiver,intentFilter)
+
+        // also handled in NETWORK_STATE_CHANGED_ACTION BroadCast
+        try {
+            val networkCallback = object : ConnectivityManager.NetworkCallback() {
+                override fun onAvailable(network: Network?) {
+                    if (!mRpcExternServerStart) return
+                    val wiFi = isconnectedToWifi()
+                    // not sure why we need a delay
+                    // but if we don't, rapidly switching the WiFi on / off doesn't work
+                    Thread.sleep(100)
+                    mRpcExternServer.update(wiFi, null)
+                }
+
+                override fun onLost(network: Network?) {
+                    if (!mRpcExternServerStart) return
+                    val wiFi = isconnectedToWifi()
+                    // not sure why we need a delay
+                    // but if we don't, rapidly switching the WiFi on / off doesn't work
+                    Thread.sleep(100)
+                    mRpcExternServer.update(wiFi, null)
+                }
+            }
+            manager = getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager?
```

Comment:
Use the [`getSystemService`](https://developer.android.com/reference/kotlin/androidx/core/content/package-summary#(android.content.Context).getSystemService()) extension function here.

Reply:
??